packages/block-editor/src/hooks/gap.js

```diff
@@ -95,18 +97,28 @@ export function GapEdit( props ) {
 		],
 	} );
 
+	const sides = useCustomSides( blockName, 'blockGap' );
+	const splitOnAxis =
+		sides && sides.some( ( side ) => AXIAL_SIDES.includes( side ) );
+
 	const ref = useBlockRef( clientId );
 
 	if ( useIsGapDisabled( props ) ) {
 		return null;
 	}
 
 	const onChange = ( next ) => {
+		let newValue = next;
+		if ( typeof next === 'object' ) {
+			const row = next?.top || 0;
+			const column = next?.left || 0;
+			newValue = row === column ? row : `${ row } ${ column }`;
```

Comment:
Why are we storing the value as a concatenated string, shouldn't we store the next object as is like we do in other kind of properties (padding, margin...)? Is it because we only allow two sides? if that's the case should we instead use a `{ vertical, horizontal }` instead if `{ top, left }` is not good enough?

Reply:
It was out of convenience to use the CSS shorthand; mainly so I didn't have to alter the [layout implemention](https://github.com/WordPress/gutenberg/blob/a010546dbb95f55ba74b0d78c27b5a8f4e394167/lib/block-supports/layout.php#L95).

There might have been other reasons, but I can't recall them. I'll revisit and confirm. 👍 

I guess it might be better in the long run if the values were separate in case we wanted to use the values independently. Though ultimately the approach wasn't right, it's where we were trying to go with https://github.com/WordPress/gutenberg/pull/35454, which added a CSS var for row and column.

We could do a check for `is_array( $gap_value )` and, in that case, declare separate `row-gap` and `column-gap` values here. 

Definitely worth another look, thanks for nudge.

> if that's the case should we instead use a { vertical, horizontal } instead if { top, left } is not good enough?

`{ top, left }` should be good enough for our purposes for now. 

The `top, right, bottom, left` prop names are baked into the [Axial Input Controls](https://github.com/WordPress/gutenberg/blob/034cd9f299fbdf6afb3dcc62ad7986df3ff5a840/packages/components/src/box-control/axial-input-controls.js) in the [BoxControl component](https://github.com/WordPress/gutenberg/blob/034cd9f299fbdf6afb3dcc62ad7986df3ff5a840/packages/components/src/box-control/index.js) anyway.

Reply:
> Definitely worth another look, thanks for nudge.

Actually, definitely worth doing now that I've looked a little more closely.  We need to support `"blockGap": [ "vertical", "horizontal" ],` and `"blockGap": [ "vertical" ],` for example. 

There is no **"doh!"** emoji 😆

Reply:
> Why are we storing the value as a concatenated string, shouldn't we store the next object as is like we do in other kind of properties (padding, margin...)? 

523d9beff8a95e1d860cda289c33b14667342f54 is a very rough first stab at it.

I'm yet to address the question of the style model. It's made a smidge easier given that we're not adding `gap` inline CSS to block containers.

But if we have to match the CSS properties, I think we'd need to use the longhand properties.


```js
	setAttributes( {
		style: {
			...style,
			spacing: {
				...style?.spacing,
                                 rowGap: next?.top || 0,
                                 columnGap: next?.left || 0,
			},
		},
	} );

```

Which probably means changing a whole bunch of other references.